### PR TITLE
refactor isBlackListed logic in ReadDataSource

### PR DESCRIPTION
**Pull request by arnaudroger**

### CODE REVIEW

File: src/main/java/demo/ReadBalancingDataSource.java [hunk 1]
```diff
@@ -32,18 +32,25 @@ public ReadDataSource(DataSource dataSource) {
...
+        public boolean isBlacklisted() {
+            long currentBlacklistedUntil;
+            do {
+                currentBlacklistedUntil = this.blacklistedUntil.get();
+
+                if (currentBlacklistedUntil == -1) {
+                    return false;
+                }
+
+                if (currentBlacklistedUntil < blacklistClockTime()) {
```
Comment (macdice) [inline src/main/java/demo/ReadBalancingDataSource.java:44]: Shoudn't this be ">" ?

File: src/main/java/demo/ReadBalancingDataSource.java [hunk 2]
```diff
@@ -32,18 +32,25 @@ public ReadDataSource(DataSource dataSource) {
...
+                    return true;
+                }
+            } while (blacklistedUntil.compareAndSet(currentBlacklistedUntil, -1)); // CAS loop
```
Comment (macdice) [inline src/main/java/demo/ReadBalancingDataSource.java:47]: Shouldn't this have "!" in front of it?  We'd only want to go around again if the CAS test fails (meaning someone else moved it).

**Comment by arnaudroger** [comment]
That's what happen when you do it between serving breakfast to the children...

File: src/main/java/demo/ReadBalancingDataSource.java (hunk 1 above)
Comment (arnaudroger) [reply src/main/java/demo/ReadBalancingDataSource.java:44]: it checked if we are still black listed so I believe it's right no?
currentBlacklistedUntil > currentTime then we are passed it ?

**Pull request merged**